src/common/sse.h

```diff
@@ -0,0 +1,127 @@
+/*
+    This file is part of darktable,
+    copyright (c) 2015 Bruce Guenter
+    darktable is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    darktable is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License
+    along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <xmmintrin.h>
+
+/*
+ * Fast SSE2 implementation of special math functions.
+ */
+
+#define POLY0(x, c0) _mm_set1_ps(c0)
+#define POLY1(x, c0, c1) _mm_add_ps(_mm_mul_ps(POLY0(x, c1), x), _mm_set1_ps(c0))
+#define POLY2(x, c0, c1, c2) _mm_add_ps(_mm_mul_ps(POLY1(x, c1, c2), x), _mm_set1_ps(c0))
+#define POLY3(x, c0, c1, c2, c3) _mm_add_ps(_mm_mul_ps(POLY2(x, c1, c2, c3), x), _mm_set1_ps(c0))
+#define POLY4(x, c0, c1, c2, c3, c4) _mm_add_ps(_mm_mul_ps(POLY3(x, c1, c2, c3, c4), x), _mm_set1_ps(c0))
+#define POLY5(x, c0, c1, c2, c3, c4, c5) _mm_add_ps(_mm_mul_ps(POLY4(x, c1, c2, c3, c4, c5), x), _mm_set1_ps(c0))
+
+#define EXP_POLY_DEGREE 4
+#define LOG_POLY_DEGREE 5
+
+/**
+ * See http://www.devmaster.net/forums/showthread.php?p=43580
+ */
+static inline __m128 _mm_exp2_ps(__m128 x)
+{
+  __m128i ipart;
+  __m128 fpart, expipart, expfpart;
+
+  x = _mm_min_ps(x, _mm_set1_ps(129.00000f));
+  x = _mm_max_ps(x, _mm_set1_ps(-126.99999f));
+
+  /* ipart = int(x - 0.5) */
+  ipart = _mm_cvtps_epi32(_mm_sub_ps(x, _mm_set1_ps(0.5f)));
+
+  /* fpart = x - ipart */
+  fpart = _mm_sub_ps(x, _mm_cvtepi32_ps(ipart));
+
+  /* expipart = (float) (1 << ipart) */
+  expipart = _mm_castsi128_ps(_mm_slli_epi32(_mm_add_epi32(ipart, _mm_set1_epi32(127)), 23));
+
+/* minimax polynomial fit of 2**x, in range [-0.5, 0.5[ */
```

Comment:
`[-0.5, 0.5]` probably?

Reply:
No idea, I didn't try to understand the code to see what ranges it's working on.